File: benchmarks/tsb/bench_index_argmin_argmax.ts
/**
 * Benchmark: index_argmin_argmax — Index.argmin and Index.argmax on 100k-element Index
 */
import { Index } from "../../src/index.js";

const SIZE = 100_000;
const WARMUP = 3;
const ITERATIONS = 20;

const labels = Array.from({ length: SIZE }, (_, i) => i);
const idx = new Index(labels);

for (let i = 0; i < WARMUP; i++) {
  idx.argmin();
  idx.argmax();
}

const start = performance.now();
for (let i = 0; i < ITERATIONS; i++) {
  idx.argmin();
  idx.argmax();
}
const total = performance.now() - start;

console.log(
  JSON.stringify({
    function: "index_argmin_argmax",
    mean_ms: total / ITERATIONS,
    iterations: ITERATIONS,
    total_ms: total,
  }),
);

File: benchmarks/pandas/bench_index_argmin_argmax.py
"""Benchmark: index_argmin_argmax — Index.argmin and Index.argmax on 100k-element Index"""
import json, time
import pandas as pd
import numpy as np

SIZE = 100_000
WARMUP = 3
ITERATIONS = 20

labels = np.arange(SIZE)
idx = pd.Index(labels)

for _ in range(WARMUP):
    idx.argmin()
    idx.argmax()

start = time.perf_counter()
for _ in range(ITERATIONS):
    idx.argmin()
    idx.argmax()
total = (time.perf_counter() - start) * 1000

print(json.dumps({
    "function": "index_argmin_argmax",
    "mean_ms": total / ITERATIONS,
    "iterations": ITERATIONS,
    "total_ms": total,
}))
